astroquery/cadc/cadctap/core.py

```diff
@@ -0,0 +1,411 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+=============
+TAP plus
+=============
+
+"""
+from astroquery.utils.tap.core import TapPlus
+from astroquery.cadc.cadctap.tapconn import TapConnCadc
+from astroquery.cadc.cadctap.jobSaxParser import JobSaxParserCadc
+import requests
+import os.path
+from astropy.extern.six.moves.urllib.parse import urlencode
+
+try:
+    # python 3
+    import http.client as httplib
+except ImportError:
+    # python 2
+    import httplib
+
+__all__ = ['TapPlusCadc']
+
+VERSION = "1.0.1"
+TAP_CLIENT_ID = "aqtappy-" + VERSION
+DEFAULT_LOGIN_URL = 'http://www.canfar.phys.uvic.ca/ac/login'
+DEFAULT_COOKIE_PREFIX = 'CADC_SSO='
+
+
+class TapPlusCadc(TapPlus):
+    """TAP plus class
+    Provides TAP and TAP+ capabilities
+    Reason for change
+    -----------------
+    To override some parts of the utils/tap library in order for it to

```

Comment:
can these changes be made in utils.tap? would that brake the gaia module? 

There has been a TAP cleanup effort by @smoh and also an open PR.
I'm not familiar with the standards and differences in the implementations to give any advice here on how to coordinate, but some coordination would definitely be would definitely welcome changes to it that makes it more generic and useable by several modules rather than everyone rolling their own.

Reply:
@bsipocz, @keflavich: The new cadc module has been started during a co-op term. My understanding is that we couldn't do the PR with our updates in the tap library because of the pending cleanup work of that module. As a temporary measure, we've forked that code in the cadc/cadctap submodule to be dropped when the tap re-work is finished. That's the reason we've kept the changes to cadctap sub-package at the minimum in spite of the fact that it does not conform with the current coding guidelines. Is there a better option?